src/rules/noUndefinedTypes.js

```diff
@@ -187,14 +187,16 @@ export default iterateJsdoc(({
   const jsdocTagsWithPossibleType = utils.filterTags(({
     tag,
   }) => {
-    return utils.tagMightHaveTypePosition(tag) && (tag !== 'suppress' || settings.mode !== 'closure');
+    return utils.tagMightHaveEitherTypeOrNamePosition(tag) && (tag !== 'suppress' || settings.mode !== 'closure');
```

Comment:
I think this makes sense as some names can be referencing, but I think we need to use a method which checks that the tags are namepath-referencing and not namepath-defining (since the latter are by definition, defined).